drivers/dma/dma_stm32.c

```diff
@@ -13,17 +13,46 @@
  *        implemented in dma_stm32_v*.c
  */
 
+#include "dma_stm32.h"
+
 #include <soc.h>
 #include <init.h>
 #include <drivers/dma.h>
 #include <drivers/clock_control.h>
 #include <drivers/clock_control/stm32_clock_control.h>
 
-#include "dma_stm32.h"
-
 #include <logging/log.h>
 LOG_MODULE_REGISTER(dma_stm32, CONFIG_DMA_LOG_LEVEL);
 
+#if defined(CONFIG_SOC_SERIES_STM32L0X) || \
+    defined(CONFIG_SOC_SERIES_STM32F0X)
+#define DMA_STM32_SHARED_IRQS
+#endif
+
```

Comment:
@gmarull @FRASTM this (the checking if IRQ's are shared) I  dislike because, would it be better to do that in Kconfig ? Or do you have any better ideas, on how to deal with the shared irq's at compile time ?

Reply:
Hi, I think if this is a IP block specific feature, we can use IP block specific kconfig symbols to do the job. We can define them in soc/arm/st_stm32/stm32xx/Kconfig.defconfig.series like the way DMA_STM32_v1 and DMA_STM32_V2 are defined. For example we define DMA_STM32_SHARED_IRQ in some socs and not in others. So in the driver context, we can do:
```
#if defined(CONFIG_DMA_STM32_SHARED_IRQ)
...
#else
...
#endif
```
But this should depend on if we do have too many socs that just using several kinds of IP blocks.

Reply:
Yes I thought about that, but didn't want to touch all SOC Kconfig files, before I knew things actually worked.  I also tried to think of some "clever" way to get the info out of the DT, but doing that at compile time with the C-preprocessor is not that easy (I might give it a second try, because the info we need is in DT).

Reply:
I just remember this info is retrieved from DT. You mean the python script? Using a function to determine if there are multiple lines use the same interrupt line, and create a new symbol in generated DT files to indicate if this one has shared IRQs? I think this surely is the clever way!

Reply:
Maybe in the future we can add a maybe-shared-flag to the `IRQ_CONNECT` macro, or make a `IRQ_CONNECT_SHARED` macro. But that is outside the scope of this PR.

For the moment I first solved it by putting it in Kconfig,  this PR is already big enough :-)